File: ServerApp/Models/Supplier.cs
using System.Collections.Generic;

namespace ServerApp.Models
{
    public class Supplier
    {
        public long SupplierId { get; private set; }
        public string Name { get; private set; }
        public string City { get; private set; }
        public string State { get; private set; }
        public IEnumerable<Product> Products { get; private set; }
    }
}

File: ClientApp/src/app/supplier.ts
import { Product } from './product';

export class Supplier {
  constructor(
    public supplierId: number,
    public name: string,
    public city: string,
    public state: string,
    public products?: Product[]) {
  }
}

export class SupplierInputModel {
  name: string;
  city: string;
  state: string;

  static fromSupplier(supplier: Supplier): SupplierInputModel {
    return {
      name: supplier.name,
      city: supplier.city,
      state: supplier.state
    };
  }
}
